src/borg/testsuite/__init__.py

```diff
@@ -54,6 +55,86 @@
     st_mtime_ns_round = -4  # only >1 microsecond resolution here?
 
 
+_symlinks_supported = None
+_fifos_supported = None
+_hardlinks_supported = None
+_utime_supported = None
+
+
+def generate_tempfile_path():
+    return os.path.join(tempfile.gettempdir(), 'borg-test-' + uuid.uuid4().hex)
```

Comment:
mktemp also opens the file (creating it). We need to not have a lock on the file, and a lot of tests also use it for non-regular files (unix sockets, hardlinks, symlinks, etc). AFAIK there's no good way to get just a file name, which is why I created this.